aten/src/ATen/Quantizer.cpp

```diff
@@ -0,0 +1,97 @@
+#include <ATen/ATen.h>
+#include <ATen/Quantizer.h>
+#include <ATen/QTensorImpl.h>
+#include <ATen/Type.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/native/TensorFactories.h>
+
+namespace at {
+
+std::shared_ptr<Quantizer> make_per_tensor_affine_quantizer(double scale, int64_t zero_point) {
+  return std::make_shared<PerTensorAffineQuantizer>(static_cast<float>(scale), static_cast<uint8_t>(zero_point));
+}
+
+QTensorImpl* get_qtensorimpl(const QTensor& self) {
+  // TODO: remove this when Variable and Tensor are merged
+  AT_ASSERTM(!self.is_variable(), "_internal_get_QTensorImpl: should not be a variable");
+  // TODO: uncomment after is_quantized() is implmeented
+  // AT_ASSERTM(self.is_quantized(), "_internal_get_QTensorImpl: not a quantized tensor");
+  return static_cast<QTensorImpl*>(self.unsafeGetTensorImpl());
+}
+
+inline QTensor new_qtensor(
+    IntArrayRef sizes, const TensorOptions& options, float scale, uint8_t zero_point) {
+  AT_ASSERT(options.device().is_cpu());
+
+  native::check_size_nonnegative(sizes);
+  auto* allocator = at::getCPUAllocator();
+  int64_t nelements = at::prod_intlist(sizes);
+  auto dtype = options.dtype();
+  AT_ASSERT(isQIntType(typeMetaToScalarType(dtype)));
+  auto storage_impl = c10::make_intrusive<StorageImpl>(
+    dtype,
+    nelements,
+    allocator->allocate(nelements * dtype.itemsize()),
+    allocator,
+    /*resizable=*/true);
+  auto quantizer = make_per_tensor_affine_quantizer(scale, zero_point);
```

Comment:
so for now we decided to hard-code quantizer type here, right? do you plan to later to pass it to new_qtensor as argument?

Reply:
Yes and Yes.

Reply:
maybe it's better to move this into quantize() function.